erpnext/controllers/accounts_controller.py

```diff
@@ -232,14 +232,19 @@ def set_taxes(self):
 
 		tax_master_doctype = self.meta.get_field("taxes_and_charges").options
 
-		if self.is_new() and not self.get("taxes"):
+		if (self.is_new() or self.update_taxes()) and not self.get("taxes"):
 			if self.company and not self.get("taxes_and_charges"):
 				# get the default tax master
 				self.taxes_and_charges = frappe.db.get_value(tax_master_doctype,
 					{"is_default": 1, 'company': self.company})
 
 			self.append_taxes_from_master(tax_master_doctype)
 
+	def update_taxes(self):
```

Comment:
The name `update_taxes` works if it is a flag. As it is a method, `is_pos_profile_changed` is better name.